sdk/containerregistry/azure-containers-containerregistry/src/test/java/com/azure/containers/containerregistry/ContainerRegistryClientAsyncIntegrationTests.java

```diff
@@ -0,0 +1,107 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+// Code generated by Microsoft (R) AutoRest Code Generator.
+
+package com.azure.containers.containerregistry;
+
+import com.azure.core.exception.ResourceNotFoundException;
+import com.azure.core.http.HttpClient;
+import com.azure.core.test.implementation.ImplUtils;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+import reactor.test.StepVerifier;
+
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import static com.azure.containers.containerregistry.TestUtils.ALPINE_REPOSITORY_NAME;
+import static com.azure.containers.containerregistry.TestUtils.DISPLAY_NAME_WITH_ARGUMENTS;
+import static com.azure.containers.containerregistry.TestUtils.HELLO_WORLD_REPOSITORY_NAME;
+import static com.azure.containers.containerregistry.TestUtils.PAGESIZE_2;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+
+public class ContainerRegistryClientAsyncIntegrationTests extends ContainerRegistryClientsTestBase {
+
+    private ContainerRegistryAsyncClient registryClient;
+    private ContainerRepositoryAsyncClient repositoryClient;
+
+    private ContainerRegistryAsyncClient getContainerRegistryAsyncClient(HttpClient httpClient) {
+        return getContainerRegistryBuilder(httpClient).buildAsyncClient();
+    }
+
+    @BeforeAll
+    static void beforeAll() {
+        StepVerifier.setDefaultTimeout(Duration.ofMinutes(30));
+        TestUtils.importImage(ImplUtils.getTestMode(), HELLO_WORLD_REPOSITORY_NAME, Arrays.asList("latest", "v1", "v2", "v3", "v4"));
+        TestUtils.importImage(ImplUtils.getTestMode(), ALPINE_REPOSITORY_NAME, Arrays.asList("latest"));
+    }
+
+    @AfterAll
+    static void afterAll() {
+        StepVerifier.resetDefaultTimeout();
+    }
+
+
+    @ParameterizedTest(name = DISPLAY_NAME_WITH_ARGUMENTS)
+    @MethodSource("getHttpClients")
+    public void listRepositories(HttpClient httpClient) {
+        registryClient = getContainerRegistryAsyncClient(httpClient);
+
+        StepVerifier.create(registryClient.listRepositories())
```

Comment:
Thoughts on this using `registryClient.listRepositories().collectList()`, as this seems what `recordWith` and `thenConsumeWhile` are attempting to perform but is a pattern that is more clear to other developers as we (I've never) see `recordWith` or `thenConsumeWhile` being used.

Reply:
I actually used this pattern for the byPage call (next few methods) as I did not want to make assumption on the order in which the paged results are returned and still wanted to check the names.

Found this to be the way to make StepVerifier collect all signals. Let me know if I missing something.